broker/broker.go

```diff
@@ -219,15 +192,83 @@ func (b *CdnServiceBroker) Update(
 		return brokerapi.UpdateServiceSpec{}, brokerapi.ErrAsyncRequired
 	}
 
-	options, err := parseUpdateDetails(details.Parameters)
+	options, err := b.parseUpdateDetails(details.Parameters)
 	if err != nil {
 		return brokerapi.UpdateServiceSpec{}, err
 	}
 
-	err = b.Manager.Update(instanceID, options.Domain, options.Origin, options.Path, options.InsecureOrigin)
+	headers := b.getHeaders(options)
+
+	err = b.manager.Update(instanceID, options.Domain, options.Origin, options.Path, options.InsecureOrigin, headers)
 	if err != nil {
 		return brokerapi.UpdateServiceSpec{}, err
 	}
 
 	return brokerapi.UpdateServiceSpec{IsAsync: true}, nil
 }
+
+// createBrokerOptions will attempt to take raw json and convert it into the "Options" struct.
+func (b *CdnServiceBroker) createBrokerOptions(details []byte) (options Options, err error) {
+	if len(details) == 0 {
+		err = errors.New("must be invoked with configuration parameters")
+		return
+	}
+	options = Options{
+		Origin: b.settings.DefaultOrigin,
+	}
+	err = json.Unmarshal(details, &options)
+	if err != nil {
+		return
+	}
+	return
+}
+
+// parseProvisionDetails will attempt to parse the update details and then verify that BOTH least "domain" and "origin"
+// are provided.
+func (b *CdnServiceBroker) parseProvisionDetails(details brokerapi.ProvisionDetails) (options Options, err error) {
+	options, err = b.createBrokerOptions(details.RawParameters)
+	if err != nil {
+		return
+	}
+	if options.Domain == "" {
+		err = errors.New("must pass non-empty `domain`")
+		return
+	}
+	if options.Origin == b.settings.DefaultOrigin {
+		if _, err = b.cfclient.GetDomainByName(options.Domain); err != nil {
+			err = fmt.Errorf("Domain %s does not exist; create it with `cf create-domain <my-organization> %s`", options.Domain, options.Domain)
```

Comment:
WDYT about putting the actual org namein the error?  It looks like the org guid is already in details so easy to lookup the name?